### ISSUE TITLE
npm run init fails running hook bootstrap: Command failed: git checkout cddd2872f0020ecf5016f326cf5e58c965eef52a

### ISSUE THREAD

**Issue by jharvell** [brave/brave-browser#6023]
## Description 

npm run init for branch 0.69.x fails

## Steps to Reproduce

   1. git checkout origin/0.69.x
   2. git reset --hard HEAD
   3. git clean -dfx
   4. git submodule update
   5. cd vendor/depot_tools
   6. git clean -dfx
   7. cd -
   8. export PATH=/opt/python2:"$PATH" // forces python2 to be used when executing python
   9. npm install
  10. npm run init 

## Actual result:

[...]
Running hooks:  95% (76/80) bootstrap                             
________ running '/usr/lib/python-exec/python2.7/python2 src/brave/script/bootstrap.py' in '/mnt/rouen/home/jharvell/brave-browser'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/style-loader-6eee6f2e/package.json'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/style-loader-6eee6f2e/CHANGELOG.md'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/style-loader-6eee6f2e/index.js'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/style-loader-6eee6f2e/LICENSE'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/style-loader-6eee6f2e/options.json'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/style-loader-6eee6f2e/README.md'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/style-loader-6eee6f2e/url.js'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/style-loader-6eee6f2e/useable.js'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/acorn-3721209c/dist/acorn.mjs'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/polished-351c48fd/docs/assets/fonts/TTF/SourceCodePro-Bold.ttf'
npm WARN tar ENOENT: no such file or directory, open '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/terser-a0c73f74/dist/bundle.min.js'
npm WARN acorn-dynamic-import@4.0.0 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.7 (node_modules/fsevents):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/fsevents-4854b31d/node_modules/abbrev' -> '/mnt/rouen/home/jharvell/brave-browser/src/brave/node_modules/.staging/abbrev-ae5d0ff1'

npm ERR! code 128
npm ERR! Command failed: git checkout cddd2872f0020ecf5016f326cf5e58c965eef52a
npm ERR! fatal: reference is not a tree: cddd2872f0020ecf5016f326cf5e58c965eef52a
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /mnt/etretat/home/jharvell/.npm/_logs/2019-09-15T19_19_03_717Z-debug.log

Traceback (most recent call last):
  File "src/brave/script/bootstrap.py", line 79, in <module>
    sys.exit(main())
  File "src/brave/script/bootstrap.py", line 33, in main
    update_node_modules('.')
  File "src/brave/script/bootstrap.py", line 69, in update_node_modules
    execute_stdout(args, env)
  File "/mnt/rouen/home/jharvell/brave-browser/src/brave/script/lib/util.py", line 187, in execute_stdout
    execute(argv, env)
  File "/mnt/rouen/home/jharvell/brave-browser/src/brave/script/lib/util.py", line 175, in execute
    raise e
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1
Error: Command '/usr/lib/python-exec/python2.7/python2 src/brave/script/bootstrap.py' returned non-zero exit status 1 in /mnt/rouen/home/jharvell/brave-browser
Hook '/usr/lib/python-exec/python2.7/python2 src/brave/script/bootstrap.py' took 34.29 secs
null
null
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! brave@0.69.125 init: `node ./scripts/sync.js --init`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the brave@0.69.125 init script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /mnt/etretat/home/jharvell/.npm/_logs/2019-09-15T19_19_05_644Z-debug.log
[2019-09-15T19_19_05_644Z-debug.log](https://github.com/brave/brave-browser/files/3614175/2019-09-15T19_19_05_644Z-debug.log)

## Expected result:
npm run init completes successfully

## Reproduces how often: 

Every time

## Brave version (brave://version info)

jharvell@wolfhound brave-browser$ git --no-pager log -n1
commit 04c066e90c7a10d76d7c443ef36ef31cdce0e1de (HEAD, origin/0.69.x)
Merge: 342ce5fda 63f470b1b
Author: Brian Clifton <brian@clifton.me>
Date:   Sat Sep 14 15:51:37 2019 -0700

    Merge pull request #6018 from brave/maxk-fix-depot-tools-0.69.x
    
    Workaround for depot_tools update on Windows (0.69.x).

## Version/Channel Information:

- Can you reproduce this issue with the current release?
- Can you reproduce this issue with the beta channel?
- Can you reproduce this issue with the dev channel?
- Can you reproduce this issue with the nightly channel?

## Other Additional Information:

I have previously successfully built 0.69.23 on this Gentoo system from the same git workspace.  Now npm run init fails for 0.69.23, but not for the same reason as 0.69.x.

Python details:
jharvell@wolfhound brave-browser$ echo $PATH
/opt/python2:/usr/lib/llvm/8/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/bin:/opt/eclipse/juno:/usr/games/bin:/opt/monero/bin:/opt/intel/vtune_amplifier/bin64
jharvell@wolfhound brave-browser$ ls -l /opt/python2
total 0
lrwxrwxrwx 1 root root 16 30 juin  12:09 python -> /usr/bin/python2
lrwxrwxrwx 1 root root 23 30 juin  12:10 python-config -> /usr/bin/python2-config
jharvell@wolfhound brave-browser$ python
Python 2.7.16 (default, Jun 23 2019, 09:31:45) 
[GCC 9.1.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> 
jharvell@wolfhound brave-browser$ python2
Python 2.7.16 (default, Jun 23 2019, 09:31:45) 
[GCC 9.1.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> 

## Miscellaneous Information:

I made sure dependencies were installed. By installing all dependencies for www-client/chromium (OS is gentoo).  Note in below command that only chromium is to be installed since all dependencies already are.
jharvell@wolfhound brave-browser$ emerge -pv www-client/chromium

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild  N     ] www-client/chromium-77.0.3865.75::gentoo  USE="closure-compile cups gnome-keyring hangouts (pic) proprietary-codecs pulseaudio suid system-ffmpeg system-icu tcmalloc -component-build -custom-cflags -jumbo-build -kerberos (-selinux) (-system-libvpx) -widevine" L10N="fr -am -ar -bg -bn -ca -cs -da -de -el -en-GB -es -es-419 -et -fa -fi -fil -gu -he -hi -hr -hu -id -it -ja -kn -ko -lt -lv -ml -mr -ms -nb -nl -pl -pt-BR -pt-PT -ro -ru -sk -sl -sr -sv -sw -ta -te -th -tr -uk -vi -zh-CN -zh-TW" 716 049 KiB

Total: 1 package (1 new), Size of downloads: 716 049 KiB

**Comment by jharvell** [comment]
Correction:  In the description I state that I had previously successfully built 0.69.23.  I checked my notes and it turns out the version I had successfully built was actually 0.69.17 (8af00a618b581da0792efeb2f0773bffac3cece3).

**Comment by jharvell** [comment]
I did get a successful run of npm run init on master.  The only difference in the procedure was that I had forgotten to put /opt/python2 at the beginning of my PATH.  npm run init failed.  Then I added /opt/python2 to the beginning of my PATH and ran npm run init.  Not sure if the modified procedure made a difference or if it was the code I was building:

commit 71cca19a87ea6bab762a2303b3cddc01e908b06f (HEAD -> master, origin/master, origin/jp-hide, origin/callback-refactor-recover-wallet, origin/callback-refactor-get-balance-report-info, origin/HEAD)
Author: Mihai PLESA <mplesa@brave.com>
Date:   Tue Sep 17 13:34:53 2019 +0100

    fixed lint (#6047)
